clang/lib/CodeGen/CGCall.cpp

```diff
@@ -4541,7 +4541,8 @@ void CodeGenFunction::EmitCallArgs(
   // evaluation, and in those cases we consider the evaluation order requirement
   // to trump the "destruction order is reverse construction order" guarantee.
   bool LeftToRight =
-      CGM.getTarget().getCXXABI().areArgsDestroyedLeftToRightInCallee()
+      (CGM.getTarget().getCXXABI().areArgsDestroyedLeftToRightInCallee() ||
+       CGM.getCodeGenOpts().EvalArgsRightToLeft)
```

Comment:
I'd write this as:

```
bool LeftToRight =
       CGM.getTarget().getCXXABI().areArgsDestroyedLeftToRightInCallee()
           ? Order == EvaluationOrder::ForceLeftToRight
           : Order != EvaluationOrder::ForceRightToLeft;
           
+  if (CGM.getCodeGenOpts().EvalArgsRightToLeft)
+    LeftToRight = Order == EvaluationOrder::ForceLeftToRight;
```

So that the downstream diff is purely additive.